components/logins/src/db.rs

```diff
@@ -183,7 +184,15 @@ impl LoginDb {
                     throw!(ErrorKind::DuplicateGuid(incoming.0.id.to_string()))
                 }
                 seen_ids.insert(incoming.0.id.clone());
-                sync_data.push(SyncLoginData::from_payload(incoming.0.clone(), incoming.1)?);
+                match SyncLoginData::from_payload(incoming.0.clone(), incoming.1) {
+                    Ok(v) => sync_data.push(v),
+                    Err(e) => {
+                        log::error!("Failed to deserialize record {:?}: {}", incoming.0.id, e);
+                        // Ideally we'd track new_failed, but it's unclear how
+                        // much value it has.
+                        telem.failed(1);
```

Comment:
Hmm, it would be nice if we had a way to report non-fatal errors like this, so we could see what it's doing. Something for Sentry, if we ever integrate it into Rust?

Reply:
I agree, it would be nice if there were a better way to report this. That said, we can probably investigate this if it turns out that we see a lot of `failed` for logins. (This is the only thing that writes that).